### ISSUE TITLE
Multiple connections from clients instead of only one

### ISSUE THREAD

**Issue by LCozzoli** [socketio/socket.io#2534]
Hello,
I've got some troubles on my sockets,
Just after updated from 1.3.7 to 1.4.5 both client and server scripts,
the problem is that few users are triggering multiple times the connection event,
I'm using an haProxy instance as middleware (latest version)

Client side connection : (the error happens without 'force new connection' too)
`_sock = io.connect('wss://myproxyserver.com:8888',{ 'force new connection': true });`

Server side:

```

var certificate = {
        key: fs.readFileSync('server.key'),
        cert: fs.readFileSync('server.crt')
    },
    app = require('express')(),
    http = require('https').Server(certificate, app),
    io = require('socket.io')(http);
...
app.get('/', function(req, res){
    res.send('');
});
...
http.listen(8888);
...
io.on('connection', function(socket){
    if (!socket.connected) return;
    console.log(socket.client.id + " == " + socket.request.headers.cookie);
...
```

Some users are well connected with only one socket, but I got few users connecting multiple time at the same moment, this users are mostly using Chrome or Mozilla, I can't reproduce the error on my browser to see if it's clientside error, but I checked socket.client.id and this multiple connections haven't the same ID, so it might not be a multiple server-side handler problem on reconnecting,
this multiples connections doesn't have the same client.id but same cookies
Have you guys an idea ?

**Comment by LCozzoli** [comment]
Seems fixed by downgrading client scripts to 1.3.7..
I really think 1.4.5 got some problems

**Comment by darrachequesne** [comment]
That issue was closed automatically. Please check if your issue is fixed with the latest release, and reopen if needed (with a [fiddle](https://github.com/darrachequesne/socket.io-fiddle) reproducing the issue if possible).
